src/Supports/BotHandler.cs

```diff
@@ -2,11 +2,14 @@
 using System.Threading.Tasks;
 using Bot.States;
 using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
 
 namespace Bot.Supports
 {
     public interface IBotHandler
```

Comment:
If we have an handler that is not a dialog we break LSP. I guess that we will need two types of chain in that case at this point. 

I propose to just document this interfaces and assume for now that  it should be used only for dialogs. Think about renaming this interface to something like `IDialogHandler` or `IBotDialogHandler`

Reply:
what do you mean an handler that is not a dialog? I thought handlers are what they are called - partial functions to handle incominc activities. Such handling could reply with a text, or trigger a dialog, or even multiple dialogs

if anything it's more of a conversation handler, in that it manages the flow of conversation delegating to chain nodes or taking priority over the chain when an handler is in the middle of a dialog

Reply:
it's just strange that we should create an empty dialog in the utility handler, we can proceed this way, but we should document it properly in the interface